web/modules/custom/weather_block/src/Plugin/Block/WeatherBlock.php

```diff
@@ -102,7 +121,7 @@ public function build(): array {
   public function getCacheTags(): array {
     $tags = [];
     $tags[] = 'config:block.block.my_awesome_theme_weatherblock';
-    $tags[] = 'user:' . \Drupal::currentUser()->id();
+    $tags[] = 'user:' . $this->accountProxy->id();
```

Comment:
You are saying that the block-level cache should be invalid at the moment when the User entity is changed.
But, does the user entity have anything to do with that block?